### PR TITLE
Vision receiver module

### PR DESCRIPTION
**Pull request by kylestach**
## Description
Refactor vision receiver into a new node and refactor to use boost::asio instead of a separate thread. ~Currently marked as draft until #1330 (which includes the Node base class) is merged.~

## Associated Issue
Issue #1329.

## Design Documents
[Link](https://docs.google.com/document/d/13LJ-HotI7wjp8G1xqDhcK5y4MAMSmWZyr55kbajW1qw/edit#heading=h.3vz8rotbuhas)

## Steps to test
This is a non-functional change. It needs QA to make sure it didn't break any of the following functionality:
 - Vision on actual field
 - Using specific vision channels on the field

### CODE REVIEW

File: soccer/Context.hpp [hunk 1]
```diff
@@ -17,4 +18,6 @@ struct Context {
     SystemState state;
     GameState game_state;
     DebugDrawer debug_drawer;
+
+    std::vector<std::unique_ptr<VisionPacket>> vision_packets;
```
Comment (kylestach) [inline soccer/Context.hpp:23]: @CollinAvidano, I think we discussed merging VisionReceiver into the VisionFilter module, and I'm now thinking that might be a good idea. That way we don't have to have this nasty vector of protobuf structures passed around and Context only contains "clean" data.

Whatever we do is pending moving logging stuff into a module, though.

**Comment by kylestach** [comment]
This is now ready for review.

**Review by JNeiger — APPROVED** [review]

**Review by SimonAbrelat — APPROVED** [review]
The changes seem totally reasonable for the VisionReceiver, but it seems weird to add more data to the already huge Context class. All that being said, I am a noob and chances are that concern is meaningless lmao.

**Comment by kylestach** [comment]
The concern is valid. Basically, context exists as a giant glob of data because we don't have proper pub/sub or similar. It's not ideal but the idea is that processor shouldn't need to know any of the particulars of the modules it calls, and they should only communicate through the Context struct.

**Comment by kylestach** [comment]
@SimonAbrelat also, see my comment on that field for future plans regarding it. Going to close this now though.

**Pull request merged**